### PR TITLE
More Operations

### PR DESCRIPTION
**Pull request by GavinK88**
Goal with this PR is to, well... add more functions.

### CODE REVIEW

File: src/mpq.h [hunk 1]
```diff
@@ -219,6 +219,136 @@ static PyObject* method_mpqobj_compact(MpqObject* self, PyObject* args) {
...
+    for (int x = 0; x < PyList_Size(flags_arg); x++) {
+
+        PyObject* list_item = PyList_GetItem(flags_arg, x);
+
+        if (unsigned int flag = get_file_verify_flag_by_alias(PyLong_AsLong(list_item))) {
+            flags |= flag;
+        }
+    }
+
+    if (flags == 0) {
```
Comment (GavinK88) [inline src/mpq.h]: move this check to before the loop

File: src/mpq.h [hunk 2]
```diff
@@ -219,6 +219,200 @@ static PyObject* method_mpqobj_compact(MpqObject* self, PyObject* args) {
...
 
+PyDoc_STRVAR(
+    docstring_method_mpqobj_sign,
+    "sign() \n--\n\n"
+    ":returns: True or False\n"
+    ":rtype: bool\n\n"
+    "Implementation of 'SFileSignArchive'. Signs the archive with a weak signature."
+    "Returns True on success, otherwise False."
+);
+static PyObject* method_mpqobj_sign(MpqObject* self, PyObject* args) {
```
Comment (whistmxl) [inline src/mpq.h]: guess we can just remove this method?

File: src/mpq.h [hunk 3]
```diff
@@ -219,6 +219,200 @@ static PyObject* method_mpqobj_compact(MpqObject* self, PyObject* args) {
...
+        return PyErr_Format(PympqBaseException, "Failed to rename file, error code: '%d'", GetLastError());
+    }
+
+    Py_RETURN_NONE;
+
+}
+
+PyDoc_STRVAR(
+    docstring_method_mpqobj_extract_file,
+    "rename_file(extract_filename, extracted_filename, search_scope /) \n--\n\n"
```
Comment (whistmxl) [inline src/mpq.h]:
```suggestion
    "extract_file(extract_filename, extracted_filename, search_scope=None /) \n--\n\n"
```

File: src/mpq.h [hunk 4]
```diff
@@ -219,6 +219,200 @@ static PyObject* method_mpqobj_compact(MpqObject* self, PyObject* args) {
...
+    ":rtype: None\n\n"
+    "Implementation of 'SFileExtractFile'. Extracts file from the MPQ."
+);
+static PyObject* method_mpqobj_extract_file(MpqObject* self, PyObject* args) {
+
+    char* extract_filename_arg = nullptr;
+    char* extracted_filename_arg = nullptr;
+    int search_scope_arg = 0;
+
+    signed int search_scope = get_file_open_flag_by_alias(search_scope_arg);
```
Comment (whistmxl) [inline src/mpq.h]: this should be done after args have been parsed, otherwise it will always be 0

File: src/mpq.h [hunk 5]
```diff
@@ -219,6 +219,200 @@ static PyObject* method_mpqobj_compact(MpqObject* self, PyObject* args) {
...
+PyDoc_STRVAR(
+    docstring_method_mpqobj_get_max_file_count,
+    "get_max_file_count() \n--\n\n"
+    ":returns: Max file count of the MPQ\n"
+    ":rtype: int\n\n"
+    "Implementation of 'SFileGetMaxFileCount'. Gets the max file count of the MPQ."
+);
+static PyObject* method_mpqobj_get_max_file_count(MpqObject* self, PyObject* args) {
+
+    int file_count = SFileGetMaxFileCount(self->hmpq);
```
Comment (whistmxl) [inline src/mpq.h:341]: I can't seem to find this stormlib function in the documentation? It's also not in this repo's README checklist.

File: src/mpq_file.h [hunk 6]
```diff
@@ -54,12 +54,26 @@ static PyObject* method_fileobj_get_name(MpqFileObject* self, PyObject* args) {
...
+PyDoc_STRVAR(
+    docstring_method_fileobj_get_size,
+    "get_size() \n--\n\n"
+    ":returns: Size of file in bytes\n"
+    ":rtype: int\n\n"
+    "Implementation of 'SFileGetFileSize'. Gets size of file."
+);
+static PyObject* method_fileobj_get_size(MpqFileObject* self, PyObject* args) {
+
+    int result = SFileGetFileSize(self->hFile, 0);
```
Comment (whistmxl) [inline src/mpq_file.h:66]: need to check if it returns `SFILE_INVALID_SIZE` and throw exception if it does

**Pull request merged**